server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
 };
 
 const removeSocket = usertoken => {
   delete sockets[usertoken];
+  delete socket_user_map[usertoken];
 };
 
-export { sockets, updateSocket, removeSocket };
+export { sockets, socket_user_map, updateSocket, removeSocket };
```

Comment:
The export statement now includes `socket_user_map`, which is a change in the module's interface. Ensure that this change is documented in the module's documentation to inform users of the new exported variable.
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*